client/src/pages/Submissions/index.js

```diff
@@ -1,39 +1,137 @@
 
 function onSubmit(data) {
 	// TODO: shwoop up to server here
 	const { submitted: _, ...rest } = data
 	console.log("Submitted!", rest)
 }
 
-export const SubmitContext = createContext()
-
 const Submissions = () => {
-	const [data, setData] = useState(initialData)
+	const [data, setData] = useState([])
+	const [images, setImages] = useState([])
 	const [active, setActive] = useState(0)
+	const dataRef = useRef()
+	const imagesRef = useRef()
+	const activeRef = useRef()
+	dataRef.current = data
+	activeRef.current = active
+	imagesRef.current = images
+
+	const newActive = (d) => {
+		for (let i = 0; i < d.length; i++) {
+			if (d[i].status === null) {
+				return i
+			}
+		}
+		return undefined
+	}
+
+	const queryData = () => {
+		httpget("/interop/odlc/list", async (response) => {
+			const odlcsEqual = (a, b) => {
+				if (a.alphanumeric === b.alphnumeric ||
+					a.alphanumeric_color === b.alphanumeric_color ||
+					a.auto_submit === b.auto_submit ||
+					a.created === b.created ||
+					a.description === b.description ||
+					a.latitude === b.latitude ||
+					a.longitude === b.longitude ||
+					a.orientation === b.orientation ||
+					a.shape === b.shape ||
+					a.shape_color === b.shape_color ||
+					a.status === b.status ||
+					a.type === b.type) {
+						return true
+					}
+				return false
+			}
+
+			let resData = response.data.result.map((odlc) => {
+				return { ...odlc, orientation: (odlc.orientation-1)*45 }
+			})
+
+			let changed = false
+			if (resData.length !== dataRef.current.length) {
+				changed = true
+			} else {
+				for (let i = 0; i < dataRef.current.length; i++) {
+					if (!odlcsEqual(dataRef.current[i], resData[i])) {
+						changed = true
+					}
+				}
+			}
+
+			if (changed) {
+				setData(resData)
+
+				let imagesData = []
+				for (let i = 0; i < resData.length; i++) {
+					await httpget("/interop/odlc/image/"+i, (res) => {
+						imagesData.push(res.data.image)
+					})
+				}
+
+				setImages(imagesData)
+			}
+		}, async (exception) => {
+			console.log(exception)
+		})
+	}
 
 	useEffect(() => {
-		data.filter(v => !v.submitted).length === 0 && setActive(undefined)
-	}, [data])
+		queryData()
+		const tick = setInterval(() => {
+			queryData()
+		}, 1000)
+		return () => clearInterval(tick)
+	}, [])
+
+	const convertOdlc = (o) => {
+		return { ...o, orientation: o.orientation/45+1 }
+	}
+
+	const accept = async (i, formData) => {
+		let d = data
+		if (formData) {
+			d[i] = formData
+			let submit = convertOdlc(d[i])
+			await httppost("/interop/odlc/edit/"+i, submit)
+		}
+		d[i].status = "submitted"
+		setData(d)
+		await httppost("/interop/odlc/submit/"+i, { status: "submitted" })
+		setActive(newActive(data))
+	}
+
+	const edit = async (i, formData) => {
+		let d = data
+		let odlc = d[i]
+		d[i] = formData
+		odlc = convertOdlc(d[i])
+		setData(d)
+		console.log(formData)
+		await httppost("/interop/odlc/edit/"+i, { ...odlc, image: images[i] })
```

Comment:
Are we not going to refresh the page after any edits? Because the images do not update after a save (although the data does carry over). I feel like refreshing the page on edit makes it much clearer for someone to know that they have changed an image.

Reply:
I forgot to change that.